binder/src/test/java/io/grpc/binder/ServerSecurityPolicyTest.java

```diff
@@ -224,24 +216,44 @@ SERVICE2, asyncPolicy((uid) -> {
                     .build();
 
     // Uses the specified policy for service1.
-    assertThat(policy.checkAuthorizationForService(MY_UID, SERVICE1).getCode())
+    assertThat(policy.checkAuthorizationForServiceAsync(MY_UID, SERVICE1).get().getCode())
             .isEqualTo(Status.INTERNAL.getCode());
-    assertThat(policy.checkAuthorizationForService(OTHER_UID, SERVICE1).getCode())
+    assertThat(policy.checkAuthorizationForServiceAsync(OTHER_UID, SERVICE1).get().getCode())
             .isEqualTo(Status.INTERNAL.getCode());
 
     // Uses the specified policy for service2.
-    assertThat(policy.checkAuthorizationForService(MY_UID, SERVICE2).getCode())
+    assertThat(policy.checkAuthorizationForServiceAsync(MY_UID, SERVICE2).get().getCode())
             .isEqualTo(Status.PERMISSION_DENIED.getCode());
-    assertThat(policy.checkAuthorizationForService(OTHER_UID, SERVICE2).getCode())
+    assertThat(policy.checkAuthorizationForServiceAsync(OTHER_UID, SERVICE2).get().getCode())
             .isEqualTo(Status.OK.getCode());
 
     // Falls back to the default.
-    assertThat(policy.checkAuthorizationForService(MY_UID, SERVICE3).getCode())
+    assertThat(policy.checkAuthorizationForServiceAsync(MY_UID, SERVICE3).get().getCode())
             .isEqualTo(Status.OK.getCode());
-    assertThat(policy.checkAuthorizationForService(OTHER_UID, SERVICE3).getCode())
+    assertThat(policy.checkAuthorizationForServiceAsync(OTHER_UID, SERVICE3).get().getCode())
             .isEqualTo(Status.PERMISSION_DENIED.getCode());
   }
 
+  /**
+   * Checks the resulting status of a {@link io.grpc.binder.ServerSecurityPolicy} built with a
+   * synchronous {@link SecurityPolicy} using both the new API and the legacy
+   * {@link ServerSecurityPolicy#checkAuthorizationForService}.
+   */
+  private static void checkSynchronousPolicy(
```

Comment:
Can you have a look at go/unit-testing-practices#assertions, especially the "single conceptual fact" part and reconsider this helper method?

Reply:
Broke it into two helpers and duplicated the tests, making it clearer when the legacy and the new APIs are exercised.

Reply:
Thanks but a reader still can't understand what each test case is actually doing without scrolling down and reading the body of the helper method. When I read the names "checkSynchronousPolicy" and "checkLegacySynchronousPolicy" I think to myself "Check what about them?"

How about if your helpers *returned* the Status that came out of the SecurityPolicy method (blocking if necessary) and tests could assert on that the same way they used to?  Truth's fluent assertions are way more readable than positional parameters on a helper method.

Reply:
I renamed the helper to have the same name as the equivalent method in the SecurityPolicy and added docs explaining it's basically a shortcut.

I dropped the synchronous helper as it doesn't feel useful under this approach anymore (it would be just a 1-line call).